a-star.ts

```diff
@@ -194,46 +211,26 @@ namespace scene {
         let curr = endDataNode;
 
         // otherwise trace back path to end
-        const output = [];
+        const output: tiles.Location[] = [];
 
         while (curr) {
-            output.unshift(curr.l);
+            output.unshift(new tiles.Location(curr.l.col, curr.l.row, tm));
             curr = curr.parent;
         }
 
         return output;
     }
 
-    function tileLocationHeuristic(tile: tiles.Location, target: tiles.Location) {
-        const startCol = locationCol(tile);
-        const startRow = locationRow(tile);
-        const endCol = locationCol(target);
-        const endRow = locationRow(target);
-
-        return ((startCol - endCol) ** 2
-            + (startRow - endRow) ** 2)
-    }
-
-    // TODO: these should probably be exposed on tiles.Location;
-    // no reason for them to be hidden
-    function locationRow(l: tiles.Location): number {
-        return l.y >> 4;
-    }
-
-    function locationCol(l: tiles.Location): number {
-        return l.x >> 4;
-    }
-
-    function isWall(l: tiles.Location, tm: tiles.TileMap) {
-        const r = locationRow(l);
-        const c = locationCol(l);
-        return tm.isObstacle(c, r);
+    function tileLocationHeuristic(tile: SimpleLocation, target: SimpleLocation) {
+        const xDist = Math.abs(target.col - tile.col)
+        const yDist = Math.abs(target.row - tile.row)
+        return Math.max(xDist, yDist) * 1000 + Math.min(xDist, yDist) * 414

```

Comment:
```suggestion
        return Math.max(xDist, yDist) * NEIGHBOR_COST + Math.min(xDist, yDist) * 
(DIAGONAL_COST - NEIGHBOR_COST)
```
update consts here also. Shall we define a DIAGONAL_MINUS_NEIGHBOR_COST? I guess not, and (DIAGONAL_COST - NEIGHBOR_COST) can be optimized by compiler, right?